src/collective/lazysizes/tests/test_lazysizes.py

```diff
@@ -69,13 +69,51 @@ def test_lazysizes_enabled_for_anonymous_user(self):
         self.assertIn(self.image.absolute_url(), img.attrib['data-src'])
         self.assertEqual(img.attrib['class'], 'lazyload')
 
-    @unittest.skipIf(IS_PLONE_5, 'Plone 4 only')
-    def test_blacklist_plone_4(self):
+    def _login(self):
+        from plone.app.testing import TEST_USER_NAME
+        from plone.app.testing import TEST_USER_PASSWORD
+        self.browser.open('http://nohost/plone/login_form')
+        self.browser.getControl(name='__ac_name').value = TEST_USER_NAME
+        self.browser.getControl(name='__ac_password').value = TEST_USER_PASSWORD
+        self.browser.getControl(name='submit').click()
+
+    def _set_registry_record(self, record, value):
         from collective.lazysizes.interfaces import ILazySizesSettings
-        record = ILazySizesSettings.__identifier__ + '.css_class_blacklist'
-        api.portal.set_registry_record(record, set(['discreet']))
+        record = ILazySizesSettings.__identifier__ + '.' + record
+        api.portal.set_registry_record(record, value)
         transaction.commit()
 
+    def test_lazysizes_disabled_for_authenticated_user(self):
+        self._login()
+        self.browser.open(self.image.absolute_url() + '/view')
+        html = lxml.html.fromstring(self.browser.contents)
+
+        # lazy loading is disabled for authenticated users
+        # main image should not be processed
+        img = html.xpath('//img')[1]  # skip first image (Plone logo)
```

Comment:
I would not use browser tests but instead test the transformer directly with a html snippet containing an image.
These tests fail on Plone 5 because there are some changes regarding the plone logo.
Also not doing browser tests should be faster.
See the tests in: https://github.com/collective/collective.lazysizes/pull/36